include/subgroup/tile/impl/load_xe.hpp

```diff
@@ -403,28 +403,29 @@ tile_load(tile_t& tile, payload_t& payload) {
   static constexpr uint32_t load_len = tile_size_x / scale_factor;
   static constexpr gpu_arch arch_tag = payload_t::arch_tag;
   using load_store_attr = load_store_attr_t<msg_type::block_1d, arch_tag>;
-  static constexpr uint32_t max_load_vec_len =
-      load_store_attr::max_load_vec_len;
+  static constexpr uint32_t max_load_vec_in_elems =
+      load_store_attr::max_load_vec_in_bytes / sizeof(load_dtype);
```

Comment:
for block_1d, 
  using mem_dtype = typename std::conditional<
      (bytes_per_row % sizeof(uint64_t) == 0) &&
          (alignment_in_bytes % sizeof(uint64_t) == 0),
      uint64_t,
      typename std::conditional<
          (bytes_per_row % sizeof(uint32_t) == 0),
          uint32_t,
          dtype>::type>::type;

max_load_vec_in_elems = load_store_attr::max_load_vec_in_bytes / sizeof(mem_dtype);